plugins/quickbuttons/__init__.py

```diff
@@ -177,7 +219,20 @@ def _add_button(self, setting: str) -> None:
                 return None
             tbs = Gtk.Button()
             tbs.set_label(_("EQ"))
-            tbs.connect("clicked", self.__on_equalizer_press)
+            tbs.connect("clicked", self._on_equalizer_press)
+
+        elif self._options[setting]["type"] == "audio_device_selection":
+            # @see plugins/previewdevice/previewprefs.py:65
+            from xl.player.gst.sink import get_devices
+
+            tbs = Gtk.ComboBoxText()
+            val = self._options[setting]["value"]
+            for device_name, device_id, _unused in list(get_devices()):
+                tbs.append(device_id, device_name)
```

Comment:
This isn't going to account for devices being added/removed. Maybe add a refresh option (right click menu?)

Also, the combo box doesn't render properly for me (I have to scroll to see the items), but I expect that's just a GTK quirk?

Reply:
You are right. I think best solution would be to connect to the combobox notify::popup-shown signal to reload the list of devices just in time.

Reply:
The rendering seems to relate to the position of the box. If it is not on the lower end of the screen it works better..